maro/rl/rl_component/rl_component_bundle.py

```diff
@@ -0,0 +1,186 @@
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_policy_creator(self) -> Dict[str, Callable[[], AbsPolicy]]:
+        """Return policy creator. Policy creator is a dictionary that contains a group of functions that generate
+        policy instances. The key of this dictionary is the policy name, and the value is the function that generate
+        the corresponding policy instance. Note that the creation function should not take any parameters.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_trainer_creator(self) -> Dict[str, Callable[[], AbsTrainer]]:
+        """Return trainer creator. Trainer creator is similar to policy creator, but is used to creator trainers.
+        """
+        raise NotImplementedError
+
+    ########################################################################################
+    # Users could overwrite the following methods                                          #
+    ########################################################################################
+    def get_device_mapping(self) -> Dict[str, str]:
+        """Return the device mapping that identifying which device to put each policy.
+
+        If user does not overwrite this method, then all policies will be put on CPU by default.
+        """
+        return {policy_name: "cpu" for policy_name in self.get_policy_creator()}
+
+    def get_policy_trainer_mapping(self) -> Dict[str, str]:
+        """Return the policy-trainer mapping which identifying which trainer to train each policy.
+
+        If user does not overwrite this method, then a policy's trainer's name is the first segment of the policy's
+        name, seperated by dot. For example, "ppo_1.policy" is trained by "ppo_1".
+
+        Only policies that provided in policy-trainer mapping are considered as trainable polices. Policies that
+        not provided in policy-trainer mapping will not be trained since we do not assign a trainer to it.
+        """
+        return {
+            policy_name: policy_name.split(".")[0] for policy_name in self.policy_names
+        }
+
+    def post_collect(self, info_list: list, ep: int) -> None:
+        """Routines to be invoked at the end of training episodes"""
+        pass
+
+    def post_evaluate(self, info_list: list, ep: int) -> None:
+        """Routines to be invoked at the end of evaluation episodes"""
+        pass
+
+    ########################################################################################
+    # Methods invisible to users                                                           #
+    ########################################################################################
+    def complete_resources(self) -> None:
+        """Generate all attributes by calling user-defined logics. Do necessary checking and transformations.
+        """
+        env_config = self.get_env_config()
+        test_env_config = self.get_test_env_config()
```

Comment:
Could these be properties of the RlBundle, and would be automatically created when being called?